src/index.js

```diff
@@ -250,9 +257,18 @@ async function replicate(stateManager, mainIndex) {
         listener.cancel();
       }
     });
-    listener.on('error', reject);
+    let lastError = undefined;
+    listener.on('error', err => {
+      lastError = err;
+      listener.cancel();
```

Comment:
On cancel does the `.drain()` is called ?

Reply:
We want to early cancel if eg 2/10 fails, we don’t want to continue with the other ones, or else it’ll skip one & be complicated